verify setup for browser (#18900)
DataDog/documentation: content/en/real_user_monitoring/connect_rum_and_traces/_index.md

[1]: /real_user_monitoring/roku/
{{< /site-region >}}

{{% /tab %}}
{{< /tabs >}}

### Verifying setup

To verify you've configured the APM integration with RUM, follow the steps below based on the SDK you installed RUM with.

{{< tabs >}}
{{% tab "Browser" %}}

1. Visit a page in your application.
2. In your browser's developer tools, go to the **Network** tab.
3. Check the request headers for a resource request that you expect to be correlated contains the [correlation headers from Datadog][1].

[1]: /real_user_monitoring/connect_rum_and_traces?tab=browserrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "Android" %}}

1. Run your application from Android Studio.
2. Visit a screen in your application.
3. Open Android Studio's [Network Inspector][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: https://developer.android.com/studio/debug/network-profiler#network-inspector-overview
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces?tab=androidrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "iOS" %}}

> 1. Run your application from XCode.

Reviewer: ```suggestion
1. Run your application from Xcode.
```

2. Visit a screen in your application.

> 3. Open XCode's [Network Connections and HTTP Traffic instrument][1].

Reviewer: ```suggestion
3. Open Xcode's [Network Connections and HTTP Traffic instrument][1].
```

{{% /tab %}}
{{% tab "React Native" %}}

1. Run your application from [software].
2. Visit a screen in your application.
3. Open [software]'s [Network Connections and HTTP Traffic instrument][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: 
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=reactnativerum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "Flutter" %}}

1. Run your application from [software].
2. Visit a screen in your application.
3. Open [software]'s [Network Connections and HTTP Traffic instrument][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: 
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=datadog#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "Roku" %}}

1. Run your application from [software].
2. Visit a screen in your application.
3. Open [software]'s [Network Connections and HTTP Traffic instrument][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: 

> [2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=datadog#how-are-rum-resources-linked-to-traces

Reviewer 2: For Roku, there's no way to easily check the outgoing requests (I've tried recently and lost 5 hours for nothing), so we can probably drop this tab

{{% /tab %}}
{{< /tabs >}}

### Verifying setup

To verify you've configured the APM integration with RUM, follow the steps below based on the SDK you installed RUM with.

{{< tabs >}}
{{% tab "Browser" %}}

1. Visit a page in your application.
2. In your browser's developer tools, go to the **Network** tab.
3. Check the request headers for a resource request that you expect to be correlated contains the [correlation headers from Datadog][1].

[1]: /real_user_monitoring/connect_rum_and_traces?tab=browserrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}

{{% tab "Mobile" %}}

1. Visit a page in your application.
2. In your browser's developer tools, go to the **Network** tab.
3. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][1].

> [1]: /real_user_monitoring/connect_rum_and_traces?tab=browserrum#how-are-rum-resources-linked-to-traces

Reviewer 2: This whole section doesn't make sense for Mobile Apps. I'm guessing the Verifying setup instructions are going to be different for each platform. E.g. for Android : 

---

1. Run your application form Android Studio
2. Visit a screen in your application.
3. Open Android Studio's [**Network Inspector**][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: https://developer.android.com/studio/debug/network-profiler#network-inspector-overview
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces?tab=androidrum#how-are-rum-resources-linked-to-traces

---

Author: Hi @xgouchet, I had a feeling that would be case, so I'm glad I asked! Would you / @DataDog/rum-mobile be able to help fill in the blanks for the remaining SDKs, as I don't have much insight into those? I did my best guess for iOS.

Reviewer 2: I'll be PTO from tomorrow until August 7th, but I'll pass the word around ;)

Author: @xgouchet That works - thanks very much! :D

[1]: /real_user_monitoring/connect_rum_and_traces?tab=browserrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "Android" %}}

1. Run your application from Android Studio.
2. Visit a screen in your application.
3. Open Android Studio's [Network Inspector][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: https://developer.android.com/studio/debug/network-profiler#network-inspector-overview
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces?tab=androidrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "iOS" %}}

1. Run your application from Xcode.
2. Visit a screen in your application.
3. Open Xcode's [Network Connections and HTTP Traffic instrument][1].
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].

[1]: https://developer.apple.com/documentation/foundation/url_loading_system/analyzing_http_traffic_with_instruments
[2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=iosrum#how-are-rum-resources-linked-to-traces

{{% /tab %}}
{{% tab "React Native" %}}

> 1. Run your application from [software].
> 2. Visit a screen in your application.
> 3. Open [software]'s [Network Connections and HTTP Traffic instrument][1].
> 4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][2].
> 
> [1]: 
> [2]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=reactnativerum#how-are-rum-resources-linked-to-traces

Reviewer 3: React Native devtools are undergoing a massive revamp, and we're currently in a state where the old devtools is unusable for a lot of projects, and the new one cannot be adopted yet by the majority of projects.

To ensure as many people can follow this and make it easy, I would recommend to use the native tools.

```suggestion
1. Run your application from XCode (respectively Android Studio).
2. Visit a screen in your application.
3. Open Xcode's [Network Connections and HTTP Traffic instrument][1] (resp. Android Studio's [Network Inspector][2]).
4. Check the request headers for a RUM resource and verify that the [required headers are set by the SDK][3].

[1]: https://developer.apple.com/documentation/foundation/url_loading_system/analyzing_http_traffic_with_instruments
[2]: https://developer.android.com/studio/debug/network-profiler#network-inspector-overview
[3]: https://docs.datadoghq.com/real_user_monitoring/connect_rum_and_traces/?tab=reactnativerum#how-are-rum-resources-linked-to-traces
```
